compiler/src/dotty/tools/dotc/transform/Inlining.scala

```diff
@@ -90,6 +90,11 @@ class Inlining extends MacroTransform, IdentityDenotTransformer {
           else super.transform(tree)
         case _: Typed | _: Block =>
           super.transform(tree)
+        case tree: UnApply =>
+          tree match
+            case tree1: UnApply if Inlines.needsInlining(tree1) =>
+              super.transform(cpy.UnApply(tree1)(fun = Inlines.inlinedUnapplyFun(tree1.fun)))
+            case tree1 => super.transform(tree1)
```

Comment:
```suggestion
        case tree: UnApply if Inlines.needsInlining(tree) =>
          super.transform(cpy.UnApply(tree)(fun = Inlines.inlinedUnapplyFun(tree.fun)))
```
Would that work? I see there is a case that should go through go through super.transform(tree) for non inline Unapply trees later as well

Reply:
I probably wanted to avoid calling `needsInlining` twice. I can refactor it to make it more concise.